### ISSUE TITLE
Version 2.48.0 broke the aws_route53_zone data lookup based on tags

### ISSUE THREAD

**Issue by mhvelplund** [terraform-providers/terraform-provider-aws#11950]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.20
+ provider.aws v2.48.0
```

### Affected Resource(s)

* aws_route53_zone (data)

### Terraform Configuration Files

```hcl
data "aws_route53_zone" "bastion" {
  provider = aws.devops
  name     = "digipro"
  vpc_id = "vpc-549f1123" # Anonymized VPC id
  tags   = map("Name", "digipro")
}
```

### Expected Behavior

In version 2.47.0 of the provider, this worked, and found the hosted zone.

### Actual Behavior

In version 2.48.0 of the provider, the code fails with the following error:

```
Error: no matching Route53Zone found

  on main.tf line 114, in data "aws_route53_zone" "bastion":
 114: data "aws_route53_zone" "bastion" {
```

### Steps to Reproduce

1. `terraform init`
1. `terraform apply`

### Important Factoids

Version 2.48.0 works if I remove the tag filter, eg. :

```hcl
data "aws_route53_zone" "bastion" {
  provider = aws.devops
  name     = "digipro"
  vpc_id = "vpc-549f1123" # Anonymized VPC id
}
```

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/pull/11661

**Comment by ewbankkit** [comment]
@mhvelplund I can reproduce this in the case that that set of tags on the data source doesn't exactly match the set of tags on the resource:

```hcl
resource "aws_route53_zone" "example" {
  tags = {
    Name = "digipro"
  }
}

data "aws_route53_zone" "example" {
  tags = {
    Name = "digipro"
  }
}
```

succeeds, but

```hcl
resource "aws_route53_zone" "example" {
  tags = {
    Name = "digipro"
    Blah = "foo"
  }
}

data "aws_route53_zone" "example" {
  tags = {
    Name = "digipro"
  }
}
```

fails with `Error: no matching Route53Zone found`.
This is a regression inadvertently introduced in the linked PR.

**Comment by rymancl** [comment]
Version 2.48.0 also broke something related to IAM permission requirements on `aws_route53_zone` when we're not even looking up by tag. We had a lot of failures this morning and had to add `"route53:ListTagsForResource"` to a policy to get around it.

**Code where the error occurs:**
```
data "aws_route53_zone" "selected" {  
  name              = var.route53_hosted_zone
  private_zone   = var.route53_is_private_zone
  vpc_id             = var.aws_vpc
}
```

**Error:**
```
Error: Error finding Route 53 Hosted Zone: AccessDenied: User: arn:aws:sts::<my-account-id>:assumed-role/<my-role>/ is not authorized to perform: route53:ListTagsForResource on resource: arn:aws:route53:::hostedzone/<my-route53-zone-id>
	status code: 403, request id: <guid>

  on .terraform\modules\my-module\main.tf line 15, in data "aws_route53_zone" "selected":
  15: data "aws_route53_zone" "selected" {  
```

Let me know if you need any additional information!

**Comment by ewbankkit** [comment]
@rymancl Yes, it looks like we are now calling `ListTagsForResource` always in order to set the `tags` attribute:

https://github.com/terraform-providers/terraform-provider-aws/blob/ba65aab7cb5d7a3d9980bf40129deebec630be66/aws/data_source_aws_route53_zone.go#L177-L185

This wasn't being done previously, so in actuality a bug was fixed by this, but we do now require additional IAM permissions.
IMHO we do not currently do the best job on specifying what IAM permissions are required for a given resource or data source and have been bitten by undocumented permission changes in the past.

**Comment by bflad** [comment]
The fix for the `tags` argument regression has been merged and will release with version 2.49.0 of the Terraform AWS Provider, Thursday this week. Thanks to @ewbankkit for the implementation. 👍
